test/run-drun/timer.mo

```diff
@@ -70,26 +70,32 @@ actor {
         id
     };
 
+    func graft(onto : ?Node, branch : ?Node) : ?Node = switch (onto, branch) {
+    case (null, null) null;
+    case (null, _) branch;
+    case (_, null) onto;
+    case (?onto, _) { ?{ onto with dopo = graft(onto.dopo, branch) } }
+    };
+
     func cancelTimer(id : TimerId) {
-        func hunt(n : ?Node) = switch n {
-          case null ();
+        func hunt(n : ?Node) : ?Node = switch n {
+          case null n;
           case (?n) { if (n.id == id) {
-                          n.expire := 0
-                      };
-                      hunt(n.ante);
-                      hunt(n.dopo)
+                          n.expire := 0;
```

Comment:
No need to do this. Since we don't iterate over this a.t.m., we can just drop `n`. Also, pattern match on `n`.

Reply:
done: 4a7cdef12